File: src/algorithms/easy/0028_Implement_strStr/ImplementStrStr.c
int strStr(char* haystack, char* needle) {
    if (needle[0] == '\0') {
        return 0;
    }

    int lengthH = strlen(haystack);
    int lengthN = strlen(needle);

    for (int i = 0; i < lengthH; i++) {
        bool skip = false;
        if (haystack[i] == needle[0] && (lengthH - i) >= lengthN) {
            for(int j = 1; j < lengthN; j++) {
                // printf("haystack: %c; needle: %c\n", haystack[i+j], needle[j]);
                // printf("Same: %d\n", (haystack[i + j] != needle[j]));
                if (haystack[i + j] != needle[j]) {
                    // printf("Break?\n");
                    skip = true;
                    break;
                }
            }

            if (skip == false) {
                return i;
            }
        }
    }

    return -1;
}

File: src/algorithms/leetcode/easy/0028_Implement_strStr/ImplementStrStr.php
<?php

class Solution
{
    /**
     * @param String $haystack
     * @param String $needle
     * @return Integer
     */
    function strStr($haystack, $needle)
    {
        if (empty($needle)) {
            return 0;
        }

        $haystackArray = str_split($haystack);
        $needleArray = str_split($needle);
        $haystackLength = sizeof($haystackArray);
        $needleLength = sizeof($needleArray);

        for ($i = 0; $i <= ($haystackLength - $needleLength); $i++) {
            $skip = false;
            if ($haystackArray[$i] == $needleArray[0]) {
                for ($j = 1; $j < $needleLength; $j++) {
                    if ($haystackArray[$i + $j] != $needleArray[$j]) {
                        $skip = true;
                        break;
                    }
                }

                if ($skip == false) {
                    return $i;
                }
            }
        }

        return -1;
    }
}

$test = new Solution;
var_dump($test->strStr("mississippi", "issip"));
